packages/cli/src/util/projects/connect-git-provider.ts

```diff
@@ -0,0 +1,101 @@
+import Client from '../client';
+import { stringify } from 'qs';
+import { Team } from '../../types';
+import chalk from 'chalk';
+
+export async function disconnectGitProvider(
+  client: Client,
+  team: Team | null,
+  projectId: string
+) {
+  const fetchUrl = `/v4/projects/${projectId}/link?${stringify({
+    teamId: team?.id,
+  })}`;
+  return client.fetch(fetchUrl, {
+    method: 'DELETE',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+}
+
+export async function connectGitProvider(
+  client: Client,
+  team: Team | null,
+  projectId: string,
+  type: string,
+  repo: string
+) {
+  const fetchUrl = `/v4/projects/${projectId}/link?${stringify({
+    teamId: team?.id,
+  })}`;
+  return client
+    .fetch(fetchUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        type,
+        repo,
+      }),
+    })
+    .catch(err => {
+      if (err.message.includes('install the GitHub integration')) {
+        client.output.error(
+          `Failed to link ${chalk.cyan(
+            repo
+          )}. Make sure there aren't any typos and that you have access to the repository if it's private.`
+        );
+      } else if (err.message.includes('connect github first')) {
+        client.output.error(
+          `Failed to link ${chalk.cyan(
+            repo
+          )}. You need to add a Login Connection to your Git provider first. Visit ${chalk.cyan(
+            `https://vercel.com/docs/concepts/personal-accounts/login-connections`
+          )} to learn how to do this.`
+        );
+      } else {
+        client.output.error(
+          `Failed to connect a Git provider repo.\nFetch url: ${fetchUrl}\n${err}`
```

Comment:
I don't think we need to log `fetchUrl` here - I don't think it would be useful to the user.